include/zephyr/net/ptp.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2022 Carbon Robotics.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <net/net_if.h>
+#include <net/ptp_time.h>
+#include <posix/time.h>
+
+#define PTP_MESSAGE_TYPE_SYNC 0x0
+#define PTP_MESSAGE_TYPE_DELAY_REQ 0x1
+#define PTP_MESSAGE_TYPE_FOLLOW_UP 0x8
+#define PTP_MESSAGE_TYPE_DELAY_RESP 0x9
+#define PTP_MESSAGE_TYPE_ANNOUNCE 0xb
+
+#define PTP_FLAG_TWO_STEP (1 << 1)
+#define PTP_FLAG_PTP_TIMESCALE (1 << 11)
+
+typedef uint8_t ptp_clock_identity[8];
+
+struct ptp_port_identity {
+	ptp_clock_identity clock_identity;
+	uint16_t port_number;
+} __packed;
+
+struct ptp_header {
```

Comment:
It would indeed be better that we avoid duplicating stuff in the header files. The gptp is a subset of ptp so creating ptp.h and using the same structs in gptp.h sounds good. Could you separate the changes to include/net/ into a separate commit and try to combine the needed gptp changes too. Perhaps that could even be a separate PR (up to you).
The sample changes shoud be in separate commits, just avoid changes to generic code in the sample commits.